src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -63,6 +63,49 @@ public void Kill()
             }
         }
 
+        /// <summary>Stops the associated process immediately.</summary>
+        private void Stop()
+        {
+            EnsureState(State.HaveNonExitedId);
+            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGSTOP) != 0)
+            {
+                throw new Win32Exception(); // to match behavior of Kill()
+            }
+        }
+
+        /// <summary>
+        /// Makes a best-effort attempt to kill all descendant (child, grandchild, etc.) processes.
+        /// </summary>
+        private void KillTree()
+        {
+            try
+            {
+                // Stop but don't kill the process. Keeps additional children from being started but leaves the process around 
+                // so that its children can be enumerated.
+                Stop();
+            }
+            catch
+            {
+                // Making a best attempt. If it fails (perhaps because the process is already dead), give up.
```

Comment:
Shouldn't we then `Kill()` it?

Reply:
We'll reach this point if the process is dead/the PID is invalid or if we don't have permissions to send kill signals. In either case, calling `Kill()` will fail.